libpod/healthcheck.go

```diff
@@ -128,12 +130,54 @@ func (c *Container) runHealthCheck() (define.HealthCheckStatus, error) {
 		hcErr = fmt.Errorf("healthcheck command exceeded timeout of %s", c.HealthCheckConfig().Timeout.String())
 	}
 	hcl := newHealthCheckLog(timeStart, timeEnd, returnCode, eventLog)
-	if err := c.updateHealthCheckLog(hcl, inStartPeriod); err != nil {
+	results, err := c.updateHealthCheckLog(hcl, inStartPeriod)
+	if err != nil {
 		return hcResult, fmt.Errorf("unable to update health check log %s for %s: %w", c.healthCheckLogPath(), c.ID(), err)
```

Comment:
It seems like we should still restart even if this fails; maybe this should just store the error, and only return it after processHealthCheckStatus is done?

Reply:
Sounds like a good idea, thanks!

Reply:
Nice catch, let's do this.